### PR TITLE
Add grouped activity feed query.

### PR DESCRIPTION
**Pull request by alstojanovic**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/alkem-io/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/alkem-io/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/alkem-io/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

### CODE REVIEW

File: src/domain/activity-feed/activity.feed.resolver.queries.ts [hunk 1]
```diff
@@ -41,4 +43,27 @@ export class ActivityFeedResolverQueries {
...
+  ): Promise<IActivityLogEntry[]> {
+    this.authorizationService.grantAccessOrFail(
+      agentInfo,
+      await this.platformAuthorizationService.getPlatformAuthorizationPolicy(),
+      AuthorizationPrivilege.READ_USERS,
+      `Activity feed query: ${agentInfo.email}`
+    );
+
+    return this.feedService.getGroupedActivityFeed(agentInfo, {
+      ...args,
```
Comment (hero101) [inline src/domain/activity-feed/activity.feed.resolver.queries.ts]: there is no need for a spread

File: src/platform/activity/activity.service.ts [hunk 2]
```diff
@@ -148,6 +148,69 @@ export class ActivityService {
...
+      defaultCondition,
+      typesCondition,
+      triggeredByCondition,
+    ]
+      .filter(condition => condition !== undefined)
+      .join(' AND ');
+
+    const groupedActivities: {
+      latest: string;
+    }[] = await this.entityManager.connection.query(
```
Comment (hero101) [inline src/platform/activity/activity.service.ts:185]: is there a reason to not use the repository class instead? - that will reduce much of the code. Also it's too complicated for the use case.

File: src/platform/activity/activity.service.ts [hunk 3]
```diff
@@ -148,6 +148,69 @@ export class ActivityService {
...
+      WHERE ${whereConditions}
+      group by activity.resourceId, activity.triggeredBy, activity.type
+      order by latest ${orderBy}
+      ${limit ? `LIMIT ${limit}` : ''};
+      `
+    );
+
+    const activityIDs = groupedActivities.map(a => a.latest);
+
+    const qb = this.activityRepository.createQueryBuilder('activity');
```
Comment (hero101) [inline src/platform/activity/activity.service.ts]: is there a reason to not use `find` instead?

File: src/domain/activity-feed/grouped.activity.feed.query.args.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import { Field, InputType } from '@nestjs/graphql';
+import { ActivityEventType } from '@common/enums/activity.event.type';
+import { UUID } from '@domain/common/scalars';
+import { ActivityFeedRoles } from './activity.feed.roles.enum';
+
+@InputType()
+export class GroupedActivityFeedQueryArgs {
```
Comment (hero101) [inline src/domain/activity-feed/grouped.activity.feed.query.args.ts]: I would suggest to reuse the existing input type as much as possible. This is not a separate query but a workaround, and when one query changes you need to change the other one.

File: src/domain/activity-feed/activity.feed.resolver.queries.ts [hunk 5]
```diff
@@ -41,4 +43,27 @@ export class ActivityFeedResolverQueries {
 
     return this.feedService.getActivityFeed(agentInfo, { ...args, pagination });
   }
+
+  @UseGuards(GraphqlGuard)
+  @Query(() => [IActivityLogEntry], {
+    nullable: false,
+    description: 'Activity events related to the current user.',
```
Comment (hero101) [inline src/domain/activity-feed/activity.feed.resolver.queries.ts]: - Please include some text about why this is a separate query.
- change the name of  the query to something like `activityFeedGrouped`. Better for searching and grouping queries.

**Review by hero101 — CHANGES_REQUESTED** [review]
also revert the grouping on the paginated query

File: src/platform/activity/activity.service.ts (hunk 2 above)
Comment (alstojanovic) [reply src/platform/activity/activity.service.ts:185]: I tried to use the activityRepository but I was not able to add a custom select for MAX(rowId). I could not retrieve the 'latest' column as the returned value is Activity.

**Review by hero101 — APPROVED** [review]

**Pull request merged**